qa/src/test/java/io/camunda/migrator/qa/runtime/element/SubprocessMigrationTest.java

```diff
@@ -51,4 +59,59 @@ public void migrateCallActivityAndSubprocess() {
         .hasVariable(LEGACY_ID_VAR_NAME, subProcessInstance.getProcessInstanceId());
     assertThat(byTaskName("userTaskName")).isCreated().hasElementId("userTaskId");
   }
+
+  @Test
+  public void shouldSkipMigrationWhenPropagateAllParentVariablesIsFalse() {
+    // given
+    deployer.deployProcessInC7AndC8("calledActivitySubprocess.bpmn");
+    deployer.deployProcessInC7AndC8("callActivityProcessNoPropagation.bpmn");
+    ProcessInstance parentInstance = runtimeService.startProcessInstanceByKey("callingProcessIdNoPropagation");
+
+    // when
+    runtimeMigrator.start();
+
+    // then
+    // verify no C8 instance was created
+    assertThatProcessInstanceCountIsEqualTo(0);
+
+    // verify the correct error message was logged
+    var events = logs.getEvents();
+    Assertions.assertThat(events.stream()
+        .filter(event -> event.getMessage()
+            .contains(String.format("Skipping process instance with legacyId [%s]: " + CALL_ACTIVITY_LEGACY_ID_ERROR,
+                parentInstance.getId(), "callActivityId"))))
+        .hasSize(1);
```

Comment:
Nit
I think this can be shortened using `logs.assertContains`:
```suggestion
    logs.assertContains(String.format("Skipping process instance with legacyId [%s]: " + CALL_ACTIVITY_LEGACY_ID_ERROR,
                parentInstance.getId(), "callActivityId"));
```